src/outlier_hub/datasets/ham10k/factory.py

```diff
@@ -0,0 +1,67 @@
+import os
+from typing import Tuple, Dict, Any
+
+from data_stack.dataset.factory import BaseDatasetFactory
+from data_stack.dataset.iterator import DatasetIteratorIF
+from data_stack.dataset.meta import IteratorMeta, MetaFactory
+from data_stack.io.storage_connectors import StorageConnector, FileStorageConnector
+
+from preprocessor import HAMPreprocessor
+from iterator import HAMIterator
+
+
+class HAMFactory(BaseDatasetFactory):
+
+    def __init__(self, storage_connector: StorageConnector):
+        self.raw_path = os.getcwd()
+        self.data_path = os.path.join(self.raw_path, "data")
+
+        super().__init__(storage_connector)
+
+    def check_exists(self) -> bool:
+        # TODO come up with a better check!
+        sample_identifier = self._get_resource_id(element="ham10k.hdf5")
+        return self.storage_connector.has_resource(sample_identifier)
+
+    def _get_resource_id(self, element: str) -> str:
+        return os.path.join(self.data_path, element)
+
+    def _prepare(self, split: str):
```

Comment:
_prepare does not need split argument

Reply:
That is true, but I left it with the thought, I will need it when I implement an advanced version, where we need **train** and **test** splits.  If it is not wanted, I will _hard-code_ it.